### PR TITLE
(Feat) Adding support for openai-compatible endpoints

### PR DESCRIPTION
**Pull request by lenaxia**
I am local hosting https://github.com/mudler/LocalAI and so need to be able to define a custom endpoint for openai. Added a checkbox to enable defining a custom URL, and clearing the config option when it is unchecked. 

I wasn't able to get the full github workflow to pass, but lint, typecheck and make all pass.

@lukasbach

### CODE REVIEW

**Comment by lenaxia** [comment]
Looks like it's struggling to find react. I'm having trouble fixing it beucase I'm not quite as familiar with typescript but will keep trying

**Comment by lenaxia** [comment]
Alright, its working now. Please feel free to test. You should be able to use an ollama instance as it provides an openai compatible endpoint

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "pensieve",
-  "version": "1.0.12",
+  "version": "1.0.13",
```
Comment (lukasbach) [inline package.json:3]: Can you please revert this? The release pipeline will bump this automatically

File: vite.base.config.ts [hunk 2]
```diff
@@ -1,19 +1,13 @@
...
   ...builtinModules.map((m) => [m, `node:${m}`]).flat(),
 ];
 
-export const external = [
-  ...builtins,
-  ...Object.keys(
-    "dependencies" in pkg ? (pkg.dependencies as Record<string, unknown>) : {},
-  ),
-];
+export const external = [...builtins];
```
Comment (lukasbach) [inline vite.base.config.ts:10]: Why was this removed?

**Review by lukasbach — COMMENTED** [review]
Hi, thank you for your contribution! Can you please have a look at the two comments I posted? 

Tests in the CI pipeline look good, you can ignore the pipeline being red (it's red because the deploy job failed, but it shouldn't deploy from PRs anyways, that's an existing config issue)

File: vite.base.config.ts (hunk 2 above)
Comment (lenaxia) [reply vite.base.config.ts:10]: This was removed because it was preventing react from being packaged in the final distro causing a "cannot find module react" error. I am using react in order to be able to utilize useEffect so that the ui can be responsive to the "useCustomUrl" switch. 

I don't think this line was doing anything meaningful anyway, as removing it does not change the final build size appreciably from what I saw.

File: package.json (hunk 1 above)
Comment (lenaxia) [reply package.json]: Done.

File: vite.base.config.ts (hunk 2 above)
Comment (lenaxia) [reply vite.base.config.ts:10]: This removal was to address my comment above:

> Looks like it's struggling to find react. I'm having trouble fixing it beucase I'm not quite as familiar with typescript but will keep trying

**Comment by lukasbach** [comment]
Thank you for your contribution!
